### PR TITLE
Added support to rename projects from project manager

### PR DESCRIPTION
**Pull request by nabor**
Link #10740

### CODE REVIEW

File: editor/project_manager.cpp [hunk 1]
```diff
@@ -443,6 +443,156 @@ class NewProjectDialog : public ConfirmationDialog {
...
+		VBoxContainer *vb = memnew(VBoxContainer);
+		add_child(vb);
+
+		Label *l = memnew(Label);
+		l->set_text(TTR("Project Path:"));
+		vb->add_child(l);
+
+		MarginContainer *mc = memnew(MarginContainer);
+		vb->add_child(mc);
+		project_path = memnew(LineEdit);
```
Comment (djrm) [inline editor/project_manager.cpp]: if this is only for renaming, why do we need this input.

File: editor/project_manager.cpp [hunk 2]
```diff
@@ -443,6 +443,156 @@ class NewProjectDialog : public ConfirmationDialog {
...
+class RenameProjectDialog : public ConfirmationDialog {
+
+	GDCLASS(RenameProjectDialog, ConfirmationDialog);
+
+private:
+	Label *error;
+	LineEdit *project_path;
+	LineEdit *project_name;
+
+	String _test_path() {
```
Comment (djrm) [inline editor/project_manager.cpp]: i dont think we need this here.

**Review by djrm — CHANGES_REQUESTED** [review]

**Comment by djrm** [comment]
@nabor after some review, there is so much repeated functionality in your PR, it might be better if you implemented something like MODE_RENAME, in the NewProjectDialog, if you want to reuse the functionality (meaning extend, the functionality of NewProjectDialog), in other case you can greatly reduce the logic following the points i suggested.

**Comment by nabor** [comment]
I was bussy but will try to reimplement this, I tried to do it with a MODE_RENAME but the code wasn't clean. I agree that there is much repeated functionality. I'll try to implement it reusing NewProjectDialog in a clener way

**Comment by nabor** [comment]
@djrm please delete this PR I'm creating a new one with your suggestions

**Comment by djrm** [comment]
@nabor you can close your own PRs, there is a button next to comment.

**Pull request closed without merging**